app/controllers/friendships_controller.rb

```diff
@@ -6,14 +6,24 @@ def create
     if @friendship.save
       flash[:success] = 'request sent'
     else
-      flash[:warning] = 'oops request failed'
+      flash[:danger] = 'oops!! request failed'
     end
-    redirect_back fallback_location @friendship
+    redirect_to users_path
+  end
+
+  def update
+    friend = Friendship.find(params[:id])
+    if friend.update_attribute(:confirmed, 'true')
+      flash[:success] = 'your info was updated'
```

Comment:
`[Optional]` for better user experience it is better you use a more descriptive message, something like: "You are now friends" or "Friendship accepted"